### PR TITLE
fixed former message

### LINKED ISSUE #181: Fix the issue of showing incorrect former message in the UI of participate, submit and my submissions tab
Behaviourally when I'm logged out and try to load a page we are first seeing `Sorry, The Challenge is not active!` message and then switching to `Please Login to access this section.` Ideally we shouldn't be seeing the former message if we are not logged-in in first place.

_Originally posted by @galipremsagar in https://github.com/Cloud-CV/EvalAI-ngx/pull/153_

Also, this issue can be seen when we reload the `submit` and `my submissions` page as well.

### PR DESCRIPTION
**Pull request by robin-gan**
Fixes #181 

#### Changes proposed in this pull request:

- Fixed the issue of showing incorrect former message in the UI of participate, submit and my submissions tab.

### CODE REVIEW

**Comment by Sanji515** [comment]
Hey @robingan7 I think you got this wrong, actually the task is fix the former message on the `participate` tab of challenge page when we reload the page. Please refer this (https://github.com/Cloud-CV/EvalAI-ngx/issues/181) issue.

**Comment by pushkalkatara** [comment]
Hi @robingan7 Please follow these steps to reproduce the issue:
1. Setting the max submission limit to 1 in a phase.
2. Try to make two submissions. An error will be shown near the submit button.
3. Now without refreshing the page select another phase and try to make a submission.
4. When you’ll select another phase, the error message shown should be cleared but it isn’t.

**Comment by robin-gan** [comment]
@pushkalkatara where can I `Setting the max submission limit to 1 in a phase`. Could you send me a route link?  Also, I don't know where to submit phase

**Comment by pushkalkatara** [comment]
Sorry, @robingan7 . Actually we confused this issue with some other issue. As @Sanji515 pointed out #181 is the issue, and the steps above are to reproduce a different issue.

**Comment by robin-gan** [comment]
@pushkalkatara I already fixed the previous one. Could you check?

File: src/app/components/challenge/challengeparticipate/challengeparticipate.component.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <div class="challenge-card">
-  <div *ngIf="!isActive" class="ev-card-panel ev-md-container card-bt-margin">
+  <div *ngIf="!isActive && isLoggedIn" class="ev-card-panel ev-md-container card-bt-margin">
```
Comment (Sanji515) [inline src/app/components/challenge/challengeparticipate/challengeparticipate.component.html:2]: @robingan7 By adding this, if challenge is not active then the message `Sorry, the challenge is not active` will not show if user is not logged in which is not the expected behaviour.

Comment (robin-gan) [reply src/app/components/challenge/challengeparticipate/challengeparticipate.component.html:2]: @Sanji515 Is it suppose to show `Please Login to access this section. ` no matter the challenge is active or not when the user is not logged in? I mean that the force login component will show up anyway.

Comment (Sanji515) [reply src/app/components/challenge/challengeparticipate/challengeparticipate.component.html:2]: > Is it suppose to show Please Login to access this section. no matter the challenge is active or not when the user is not logged in

Yes, if this is the case then putting `isLoggedIn` condition is correct. In the current version it check for only if the challenge is active or not and shows the message `Sorry, the challenge is not active` which I think should not happen. The flow should be like first user will login and then see the challenge participation status. So I think putting `isLoggedIn` condition is correct :+1:

**Review by Sanji515 — APPROVED** [review]
Thanks for the fix @robingan7 , LGTM! This can be merged :tada:

**Review by RishabhJain2018 — APPROVED** [review]

**Pull request merged**